### PR TITLE
[2938] cherry pick aec_chain_sim improvements

### PR DESCRIPTION
**Pull request by wj2016**

### CODE REVIEW

**Review by cytadela8 — CHANGES_REQUESTED** [review]
Missed changes from commit  cd101d91c.

**Comment by cytadela8** [comment]
For future refference purposes. Changes in apps/aechannel/test/aesc_chain_watcher_SUITE.erl are from 19cd96dfa9.

**Comment by wj2016** [comment]
> For future refference purposes. Changes in apps/aechannel/test/aesc_chain_watcher_SUITE.erl are from [19cd96d](https://github.com/aeternity/aeternity/commit/19cd96dfa9aa919c0329eb539c1f27fd2775b631).

[19cd96d] Upgrade chain simulator (#3358) in case the commit id changes in the future.

File: apps/aecore/test/aec_chain_sim.erl [hunk 1]
```diff
@@ -664,14 +661,14 @@ new_channel(#{ channel_id       := _ChId
              , channel_reserve  := ChanReserve
              , lock_period      := LockPeriod
              , state_hash       := StateHash
-             , nonce            := Nonce
-             , delegate_ids     := Delegates } = _CreateTx, Protocol) ->
+             , nonce            := Nonce } = _CreateTx) ->
+    Protocol = 5,    %% LIMA protocol, not that it's likely to matter here
```
Comment (cytadela8) [inline apps/aecore/test/aec_chain_sim.erl]: Please use the define from aecontract/include/hard_forks.hrl

**Review by cytadela8 — APPROVED** [review]

**Review by velzevur — APPROVED** [review]

**Review by uwiger — APPROVED** [review]

**Pull request merged**